src/boot/efi/boot.c

```diff
@@ -2155,6 +2193,96 @@ static void config_entry_add_linux(
                         }
                 }
         }
+
+        uefi_call_wrapper(linux_dir->Close, 1, linux_dir);
+}
+
+static EFI_STATUS secure_boot_enroll(EFI_FILE *root_dir, Config *config, ConfigEntry *entry) {
+        assert(root_dir);
+        assert(config);
+        assert(entry);
+
+        typedef struct {
+                const CHAR16 *name;
+                const CHAR16 *filename;
+                const EFI_GUID vendor;
+                CHAR8 *buffer;
+                UINTN size;
+        } SecureBootVariable;
```

Comment:
You are leaking several `buffer`s. And honestly, I don't think this struct is really necessary...